bitrise.yml

```diff
@@ -6,31 +6,50 @@ trigger_map:
   - push_branch: main
     pipeline: app-store-pipeline
   - pull_request_target_branch: '*'
-    pipeline: build-test-pipeline
+    pipeline: qa-build-test-pipeline
 
 pipelines:
-  build-test-pipeline:
+  # Will create a QA Build and run a test across multiple simulator devices
+  qa-build-test-pipeline:
     stages:
-      # Run the buld and test stage workflows
-      - build-test: {}
+      # Build a qa build, and build a test build in parallel
+      - build-and-test: {}
+      # Run multiple tests on simulators in parallel
+      - run-test-groups: {}
+      - deploy-test-results: {}
+  # Will create an app store build and upload it to itunes connect
   app-store-pipeline:
     stages:
       # First Build the release build (app store build)
       - release-build: {}
       # Run the steps to deploy to the app store
       - deploy-app-store: {}
+  # Will create an app store capabale build, but release it to Internal Testing users in Testflight.
+  # Intended to run nightly.
   nightly-internal-pipeline:
     stages:
       # First Build the release build (app store build)
       - release-build: {}
       # Run the steps to deploy to testflight internally
       - deploy-internal-testflight: {}
 stages:
-  build-test:
+  build-and-test:
     workflows:
-      # Run the test and qa build flow in parallel
-      - test: {}
+      # Build a test build
+      - test-build: {}
+      # Build a qa build
       - qa-build: {}
```

Comment:
For the qa-build I would also suggest using another stack, the limitation here is the test-build that takes longer and this workflow is not running tests so it could use andother cheaper stack.

Reply:
Done! We now use the standard machines as default except for UI tests